pkg/controller/servicebinding/reconciler.go

```diff
@@ -157,16 +170,55 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		r.restMapper,
 	)
 	if err != nil {
+
+		_ = updateSBRConditions(r.dynClient, sbr,
+			conditionsv1.Condition{
+				Type:    CollectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+			conditionsv1.Condition{
+				Type:    InjectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+			conditionsv1.Condition{
+				Type:    BindingReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+		)
 		return requeueError(err)
 	}
-
 	binding, err := buildBinding(
 		r.dynClient,
 		sbr.Spec.CustomEnvVar,
 		serviceCtxs,
 		sbr.Spec.EnvVarPrefix,
 	)
 	if err != nil {
+		_ = updateSBRConditions(r.dynClient, sbr,
+			conditionsv1.Condition{
+				Type:    CollectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+			conditionsv1.Condition{
+				Type:    InjectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			},
+			conditionsv1.Condition{
+				Type:    BindingReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  FailedToGatherServiceInformationReason,
+				Message: err.Error(),
+			})
```

Comment:
same comments as above. Error in `buildServiceContexts` or `buildBinding` does not move the process into the same condition.